codegenerator/cli/templates/static/codegen/src/Utils.res

```diff
@@ -43,16 +43,13 @@ module Dict = {
    */
   external dangerouslyGetNonOption: (dict<'a>, string) => option<'a> = ""
 
-  let merge = (a: dict<'a>, b: dict<'a>) => {
-    let result = Js.Dict.empty()
-    Js.Dict.entries(a)->Js.Array2.forEach(((key, value)) => {
-      result->Js.Dict.set(key, value)
-    })
-    Js.Dict.entries(b)->Js.Array2.forEach(((key, value)) => {
-      result->Js.Dict.set(key, value)
-    })
-    result
-  }
+  let merge: (dict<'a>, dict<'a>) => dict<'a> = %raw(`(dictA, dictB) => ({...dictA, ...dictB})`)
+
+  let updateImmutable: (
+    dict<'a>,
+    string,
+    'a,
+  ) => dict<'a> = %raw(`(dict, key, value) => ({...dict, [key]: value})`)
```

Comment:
I added these raw js functions as I thought that would be the most performant way to do these two operations.

You can see what merge was doing before and I think spreads would be most performant.

Reply:
As I know spreads have terrible performance, so it's not guaranteed for them to be faster. But in the case probably it will be.

Reply:
For this case, I'd just kept it everything as it is. I only have a comparison data with inlining the field set https://x.com/artalar_dev/status/1571839333694324736

Reply:
For Object.assign need to do a benchmark, but I doubt there will be a difference which is worth to consider